lib/xdrgen/generators/rust.rb

```diff
@@ -94,6 +95,26 @@ def render_lib(out)
         out.break
       end
 
+      def render_type_macro(out, types)
+        out.puts <<-EOS.strip_heredoc
+        #[doc(hidden)]
+        #[macro_export]
+        macro_rules! _call_macro_with_each_type_#{@output.inputs_hash} {
+            // The x-macro takes a single ident, the name of a macro to call ...
+            ($macro_to_call_back:ident, $($context:tt),*) => {{
+                // ... and calls it back, once for each XDR type.
+                #{types.map do |t|
+                <<-EOS
+                $macro_to_call_back!(#{t}, $($context),*);
+                EOS
+                end.join("\n")}
+
+            }};
+        }
+        pub use _call_macro_with_each_type_#{@output.inputs_hash} as call_macro_with_each_type;
+        EOS
+      end
```

Comment:
I forgot there's an issue with this that I need to tweak before merging this. I need to replace the inputs hash with a configurable name to pass in.